### PR TITLE
WIP: Allow logstash prefix via key in record

### PR DESCRIPTION
**Pull request by portante**
DESCRIPTION HERE

This is a Work-in-Progress to test out a simple behavior change to obviate the need for elasticsearch_dynamic in OpenShift's origin-aggregated-loggings fluentd use.

Basically, we only need to be able to specify the index prefix dynamically. This can be provided by a simple key in the record being indexed.

It avoids the overhead CPU calculations necessary to eval all the dynamic parameters.

This is untested, and purely a proof of concept to demonstrate a possibility.

(check all that apply)
- [ ] tests added
- [ ] tests passing
- [ ] README updated (if needed)
- [ ] README Table of Contents updated (if needed)
- [ ] History.md and `version` in gemspec are untouched
- [ ] backward compatible
- [ ] feature works in `elasticsearch_dynamic` (not required but recommended)

### CODE REVIEW

File: lib/fluent/plugin/out_elasticsearch.rb [hunk 1]
```diff
@@ -88,6 +89,10 @@ def configure(conf)
       @remove_keys_on_update = @remove_keys_on_update.split ','
     end
 
+    if @logstash_prefix_key && @logstash_prefix_key.is_a?(String)
+      @logstash_prefix_key = @logstash_prefix_key.split '.'
```
Comment (richm) [inline lib/fluent/plugin/out_elasticsearch.rb]: What does this do?

Comment (portante) [inline lib/fluent/plugin/out_elasticsearch.rb]: Not sure exactly, just copied what `target_index_key` did.

Comment (richm) [inline lib/fluent/plugin/out_elasticsearch.rb]: `@target_index_key` can be in the form "parentkey.childkey" https://github.com/uken/fluent-plugin-elasticsearch/blob/master/lib/fluent/plugin/out_elasticsearch.rb#L291

If `@logstash_prefix_key` should also be able to handle parent/child nesting, then similar logic is needed.

**Comment by portante** [comment]
Working fixes through ViaQ project first.

**Pull request closed without merging**